pkg/asset/cluster/azure/azure.go

```diff
@@ -30,13 +31,60 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 // PreTerraform performs any infrastructure initialization which must
 // happen before Terraform creates the remaining infrastructure.
 func PreTerraform(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
-	if len(installConfig.Config.Azure.ResourceGroupName) == 0 {
+	session, err := installConfig.Azure.Session()
+	if err != nil {
+		return fmt.Errorf("failed to get session: %q", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to get session: %w", err)
```
Use the `%w` verb when wrapping errors.

Reply:
done